### PR TITLE
Itunes test

### PR DESCRIPTION
**Pull request by aliarslanansari**
### Ticket Link
---------------------------------------------------

### Related Links
---------------------------------------------------

### Description
---------------------------------------------------

### Steps to Reproduce / Test
---------------------------------------------------

### GIF's
---------------------------------------------------

### CODE REVIEW

File: app/components/Header/index.js [hunk 1]
```diff
@@ -39,7 +39,7 @@ function Header(props) {
   return (
     <StyledHeader {...props} data-testid="header">
       <Logo alt="logo" src={logo} />
-      <Title type="heading" id="wednesday_solutions" />
+      <Title type="heading" id="itunesmusic" />
```
Comment (tapan0212) [inline app/components/Header/index.js:42]: Refer this for variable nomenclature we follow
https://playbook.wednesday.is/engineering-playbook/the-wednesday-way/react

File: app/containers/ItunesContainer/index.js [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+      dispatchClearItunesSongs();
+    }
+  };
+  const debouncedHandleOnChange = debounce(handleOnChange, 200);
+
+  const renderRepoList = () => {
+    const items = get(songsData, 'results', []);
+    const resultCount = get(songsData, 'resultCount', 0);
+    return (
+      (items.length !== 0 || loading) && (
```
Comment (tapan0212) [inline app/containers/ItunesContainer/index.js:87]: We can use If component from `@components/If`

File: app/containers/ItunesContainer/index.js [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+              <div>
+                <T id="search_query" values={{ songName }} />
+              </div>
+            )}
+            {resultCount !== 0 && (
+              <div>
+                <T id="matching_results" values={{ resultCount }} />
+              </div>
+            )}
+            {items.map((item, index) => (
```
Comment (tapan0212) [inline app/containers/ItunesContainer/index.js:100]: We can use For component from `@components/For`

File: app/containers/ItunesContainer/reducer.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
+/*
+ *
+ * HomeContainer reducer
```
Comment (tapan0212) [inline app/containers/ItunesContainer/reducer.js:3]: Make use of react floki

File: app/containers/ItunesContainer/index.js [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+                <T id="release_date" values={{ stars: item.releaseDate }} />
+              </CustomCard>
+            ))}
+          </Skeleton>
+        </CustomCard>
+      )
+    );
+  };
+  const renderErrorState = () => {
+    let repoError;
```
Comment (tapan0212) [inline app/containers/ItunesContainer/index.js:113]: repoErorr?

File: internals/webpack/webpack.base.babel.js [hunk 6]
```diff
@@ -163,8 +163,7 @@ module.exports = options => ({
       '@themes': path.resolve(__dirname, '../../app/themes'),
       '@images': path.resolve(__dirname, '../../app/images'),
       '@hooks': path.resolve(__dirname, '../../app/hooks'),
-      moment$: path.resolve(__dirname, '../../node_modules/moment/moment.js'),
-      '@ant-design/icons/lib/dist$': path.resolve(__dirname, './app/icons.js')
```
Comment (tapan0212) [inline internals/webpack/webpack.base.babel.js:167]: what was the purpose of removing this?

File: app/containers/ItunesContainer/index.js [hunk 7]
```diff
@@ -0,0 +1,199 @@
...
+    }
+  }, [songsData]);
+
+  useEffect(() => {
+    if (songName && !songsData?.results?.length) {
+      dispatchItunesSongs(songName);
+      setLoading(true);
+    }
+  }, []);
+  useInjectSaga({ key: 'itunesContainer', saga });
```
Comment (tapan0212) [inline app/containers/ItunesContainer/index.js:69]: Refer this for expected code order
[React Process.pdf](https://github.com/wednesday-solutions/react-template/files/5187495/React.Process.pdf)

**Review by tapan0212 — CHANGES_REQUESTED** [review]

File: app/components/Header/index.js (hunk 1 above)
Comment (aliarslanansari) [reply app/components/Header/index.js:42]: Thanks, will follow that from now onwards

File: app/containers/ItunesContainer/index.js (hunk 2 above)
Comment (aliarslanansari) [reply app/containers/ItunesContainer/index.js:87]: Ok ill check it out

File: app/containers/ItunesContainer/index.js (hunk 7 above)
Comment (aliarslanansari) [reply app/containers/ItunesContainer/index.js:69]: Thanks, will follow the order

File: internals/webpack/webpack.base.babel.js (hunk 6 above)
Comment (aliarslanansari) [reply internals/webpack/webpack.base.babel.js:167]: I didn't removed it, maybe its removed while building

File: app/containers/ItunesContainer/index.js (hunk 5 above)
Comment (aliarslanansari) [reply app/containers/ItunesContainer/index.js:113]: I'll update it

File: app/containers/ItunesContainer/index.js (hunk 3 above)
Comment (aliarslanansari) [reply app/containers/ItunesContainer/index.js:100]: OK will do that

File: app/containers/ItunesContainer/reducer.js (hunk 4 above)
Comment (aliarslanansari) [reply app/containers/ItunesContainer/reducer.js:3]: Sorry i didn't got it, can you elaborate

**Pull request closed without merging**